basic_module/account/src/traits/impls.rs

```diff
@@ -0,0 +1,150 @@
+                        let _ = sub_balance(&sender, tx.fee + quantity);
+                        let _ = add_balance(&receiver, *quantity);
+                    }
+                }
+                Action::Custom {
+                    ..
+                } => {}
+            }
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct Reviser {}
+
+impl AccountReviser for Reviser {
+    fn add_balance(address: &Address, val: u64) -> Result<(), AccountError> {
+        let context = get_context();
+        let mut account: Account = context.get(get_storage_id(), address).ok_or(AccountError::InvalidAddress)?.into();
+
+        account.balance += val;
+        context.set(get_storage_id(), address, account.to_vec());
+        Ok(())
+    }
+
+    fn sub_balance(address: &Address, val: u64) -> Result<(), AccountError> {
+        let context = get_context();
+        let mut account: Account = context.get(get_storage_id(), address).ok_or(AccountError::InvalidAddress)?.into();
+
+        if account.balance < val {
+            return Err(AccountError::InvalidValue)
+        }
+
+        account.balance -= val;
+        context.set(get_storage_id(), address, account.to_vec());
+        Ok(())
+    }
+
+    fn set_balance(address: &Address, val: u64) -> Result<(), AccountError> {
+        let context = get_context();
+        let mut account: Account = context.get(get_storage_id(), address).ok_or(AccountError::InvalidAddress)?.into();
+
+        account.balance = val;
+        context.set(get_storage_id(), address, account.to_vec());
+        Ok(())
+    }
+
+    fn increase_sequence(address: &Address) -> Result<(), AccountError> {
+        let context = get_context();
+        let mut account: Account = context.get(get_storage_id(), address).ok_or(AccountError::InvalidAddress)?.into();
+
+        account.sequence += 1;
+        context.set(get_storage_id(), address, account.to_vec());
+        Ok(())
+    }
+}
+
+#[allow(dead_code)]
+pub struct View {}
+
+impl AccountView for View {
+    fn has_account(address: &Address) -> bool {
+        get_context().has(get_storage_id(), address)
+    }
+
+    fn is_active(address: &Address) -> Result<bool, AccountError> {
+        let is_active: Option<bool> = get_context()
+            .get(get_storage_id(), address)
+            .map(|_| Account {
+                sequence: 0,
+                balance: 0,
+            })
+            .map(|account| account.balance != 0 || account.sequence != 0);
+        is_active.ok_or(AccountError::InvalidAddress)
+    }
+
+    fn get_balance(context: &mut dyn SubStorageAccess, address: &Address) -> Option<u64> {
+        context
+            .get(get_storage_id(), address)
+            .map(|_| Account {
+                sequence: 0,
+                balance: 0,
+            })
+            .map(|account| account.balance)
+    }
+
+    fn get_sequence(context: &mut dyn SubStorageAccess, address: &Address) -> Option<u64> {
+        context
```

Comment:
Why don't you reuse the module scope function already implemented in the `executer`?